docs/rfcs/033-pageserver-postgres-bundle.md

```diff
@@ -0,0 +1,139 @@
+# Postgres Bundle for Pageserver
+
+Created on 2024-06-17
+
+## Summary
+
+This RFC defines the responsibilities of Compute and Storage team regarding the
+build & deployment of the Postgres code that Pageserver must run
+(`initdb`, `postgres --wal-redo`).
+
+## Motivation
+
+Pageserver has to run Postgres binaries to do its job, specifically
+
+* `initdb`
+* `postgres --wal-redo` mode
```

Comment:
We also can build a separate binary in theory and avoid linking it with random things. We already had a case of excessive memory usage in wal-redo because we allocated some hashmaps that are used in compute but are not needed in wal-redo. Some pass over the code to set missing `if (i-am-wal-redo)` would be helpful